core/registry/cache.go

```diff
@@ -69,3 +78,33 @@ func SetNoIndexCache() { MicroserviceInstanceIndex = newNoIndexCache() }
 
 // newCacheIndex returns index implemention according to config
 func newCacheIndex() CacheIndex { return newIndexCache() }
+
+// get local provider cache
+func GetProvidersFromCache() []*MicroService {
+	microServices := make([]*MicroService, 0, 512)
+	items := ProvidersMicroServiceCache.Items()
+	for key := range items {
+		keys := strings.Split(key, "|")
+		if len(keys) != 2 {
+			lager.Logger.Warn("split cache item key lengths is not 2 ")
+			continue
+		}
+		microServices = append(microServices, &MicroService{ServiceName: keys[0], AppID: keys[1], Version: common.AllVersion})
+	}
+	return microServices
+}
+
+// refresh provider cache
+func AddProviderToCache(serverName, appID string) {
+	if appID == "" {
+		appID = config.GetGlobalAppID()
+		if appID == "" {
+			appID = common.DefaultApp
+		}
+	}
+	key := strings.Join([]string{serverName, appID}, "|")
+	_, ok := ProvidersMicroServiceCache.Get(key)
+	if !ok {
+		ProvidersMicroServiceCache.Set(key, struct{}{}, 0)
```

Comment:
doesn't make sense to use struct{}{}, plz use real struct here
it should has name and app id